### PR TITLE
Added assertion methods toHaveLength & toBeType

### PR DESCRIPTION
**Pull request by carlos-ds**

### CODE REVIEW

**Comment by carlos-ds** [comment]
- The deepcode minor issues were already applicable for the existing code.
- Once this PR is merged, I'd also like to update the Wiki but I guess that is only available for collaborators, or am I missing something? 
- I looked into writing tests for all the custom assertion methods but I left it out of scope here because it's not that easy. I might look it into it further and open another PR for that.

File: helpers/postwomanTesting.js [hunk 1]
```diff
@@ -144,4 +144,40 @@ class Expectation {
...
+    const actualLength = this.expectValue.length
+    return this._satisfies(actualLength, expectedLength)
+      ? this._pass()
+      : this._fail(
+          this._fmtNot(
+            `Expected length to be ${expectedLength} but actual length was ${actualLength}`
+          )
+        )
+  }
+  toBeOfType(expectedType) {
```
Comment (liyasthomas) [inline helpers/postwomanTesting.js:157]:
```suggestion
  toBeType(expectedType) {
```

File: helpers/postwomanTesting.js [hunk 2]
```diff
@@ -144,4 +144,40 @@ class Expectation {
...
+        "object",
+        "undefined",
+        "bigint",
+        "symbol",
+        "function",
+      ].indexOf(expectedType) < 0
+    ) {
+      return this._fail(
+        this._fmtNot(
+          `Argument for toBeOfType should be "string", "boolean", "number", "object", "undefined", "bigint", "symbol" or "function"`
```
Comment (liyasthomas) [inline helpers/postwomanTesting.js:173]:
```suggestion
          `Argument for toBeType should be "string", "boolean", "number", "object", "undefined", "bigint", "symbol" or "function"`
```

**Review by liyasthomas — CHANGES_REQUESTED** [review]

File: helpers/postwomanTesting.js (hunk 1 above)
Comment (liyasthomas) [reply helpers/postwomanTesting.js:157]: `toBeType` sounds much better than `toBeOfType`

File: helpers/postwomanTesting.js (hunk 2 above)
Comment (liyasthomas) [reply helpers/postwomanTesting.js:173]: `toBeType` sounds much better than `toBeOfType`

**Comment by liyasthomas** [comment]
@carlos-ds thanks for the quick PR. I've updated the [Wiki page](https://github.com/hoppscotch/hoppscotch/wiki/Post-Requests-Tests) with [`.toBeType(type)`](https://github.com/hoppscotch/hoppscotch/wiki/Post-Requests-Tests#tobetypetype) and [`toHaveLength(number)`](https://github.com/hoppscotch/hoppscotch/wiki/Post-Requests-Tests#tohavelengthnumber) entries.

**Comment by liyasthomas** [comment]
Squash and merging PR.

**Review by liyasthomas — APPROVED** [review]
LGTM.

**Comment by liyasthomas** [comment]
Unit tests for test scripts are much appreciated - let us know if you need any help with that.

File: helpers/postwomanTesting.js (hunk 1 above)
Comment (carlos-ds) [reply helpers/postwomanTesting.js]: Well I'm not a native English speaker but "to be of type" would be grammatically correct I believe. But as long as it's clear in the docs I guess it doesn't matter.

**Comment by carlos-ds** [comment]
@liyasthomas yes please, if you can grant me rights to edit the wiki then I could also help in that department.

**Comment by liyasthomas** [comment]
@carlos-ds Let's stick to `toBeType` for now. Also I've given access to edit Wiki. Let me know once you're done with Wiki edits.

**Pull request merged**